src/containers/CircleCI/List.js

```diff
@@ -0,0 +1,100 @@
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Redirect } from 'react-router';
+import { Link } from 'react-router-dom';
+
+import moment from 'moment'
+import uuidv1 from 'uuid'
+
+import {
+  getBuildList,
+  selectBuild,
+} from '../../reducers/CircleCIReducer/actions';
+
+import './css/List.css'
+
+class List extends Component {
+
+  state = { 
+    redirect: false, 
+  }
+
+  componentDidMount() {
+    const apiKey = sessionStorage.getItem('CIRCLE_CI_API_KEY');
+    this.props.getBuildList(apiKey);
+    this.setState({redirect: !apiKey});
+  }
+
+  selectBuild = (build) => () => {
+    this.props.selectBuild(build);
+  }
+
+  render() {
+    if(this.state.redirect || this.props.error) return <Redirect to='/api_key' />
+    return (
+      <div className="builds_container">
+        { this.props.dataFetched ? 
+          <div>
+            <h1 className="builds_container__header">Circle CI</h1>
+            { this.renderLatestBuilds() }
+          </div> :
+          <div>Loading data...</div>
+        }
+      </div>
+    );
+  }
+
+  renderLatestBuilds() {
+    return (
+      <div>
+        <h2>Latest builds</h2>
+        {
+          this.props.list.map(build => {
+            return this.renderSingleBuild(build)
+          })
+        }
+      </div>
+    )
+  }
+
+  renderSingleBuild(build) {
+    return (
+      <div key={uuidv1()} className="single-build">
```

Comment:
This should be a separate component

Reply:
Also, don't use uuidv1 for key as it always returns other key, meaning that the whole list will always be rerendered when render is called, even if there's no need for that:
key should be unique, yet persistent per item